src/Ultz.SuperInvoke.Loader/LibraryLoader.cs

```diff
@@ -69,18 +124,37 @@ public IntPtr LoadNativeLibrary(string[] names, PathResolver pathResolver)
             if (names == null || names.Length == 0)
                 throw new ArgumentException("Parameter must not be null or empty.", nameof(names));
 
-            var ret = IntPtr.Zero;
+            var success = TryLoadNativeLibrary(names, pathResolver, out IntPtr result);
+
+            if (!success)
+                throw new FileNotFoundException($"Could not find or load the native library from any name: [ {string.Join(", ", names)} ]");
+
+            return result;
+        }
+
+        /// <summary>
+        ///     Loads a native library by name and returns an operating system handle to it.
+        /// </summary>
+        /// <param name="names">
+        ///     An ordered list of names. Each name is tried in turn, until the library is successfully loaded.
+        /// </param>
+        /// <param name="pathResolver">The path resolver to use.</param>
+        /// <param name="result">A pointer to the loaded library.</param>
+        /// <returns>The operating system handle for the shared library.</returns>
+        public bool TryLoadNativeLibrary(string[] names, PathResolver pathResolver, out IntPtr result)
+        {
+            result = IntPtr.Zero;
+
+            if (names == null || names.Length == 0)
+                return false;
+
             foreach (var name in names)
             {
-                ret = LoadWithResolver(name, pathResolver);
-                if (ret != IntPtr.Zero) break;
+                result = LoadWithResolver(name, pathResolver);
```

Comment:
This still uses the throwing method when it should use the try-style method (in its current form the first chance exceptions we sought to fix are still thrown!)

Reply:
Are you talking about the **TryLoadNativeLibrary** method on line 144?  I do not see any exceptions being thrown in there.

Reply:
Ah whoops, sorry I reviewed this on my phone late last night. This is all fine turns out.